x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/drilldown.tsx

```diff
@@ -46,7 +53,70 @@ export class DashboardToDashboardDrilldown
     return true;
   };
 
-  public readonly execute = () => {
-    alert('Go to another dashboard!');
+  public readonly execute = async (
+    config: Config,
+    context: ActionContext<VisualizeEmbeddableContract>
+  ) => {
+    const getUrlGenerator = await this.params.getGetUrlGenerator();
+    const navigateToApp = await this.params.getNavigateToApp();
+
+    const {
+      createFiltersFromRangeSelectAction,
+      createFiltersFromValueClickAction,
+    } = await this.params.getDataPluginActions();
+    const {
+      timeRange: currentTimeRange,
+      query,
+      filters: currentFilters,
+    } = context.embeddable!.getInput();
+
+    // if useCurrentDashboardFilters enabled, then preserve all the filters (pinned and unpinned)
+    // otherwise preserve only pinned
+    const existingFilters =
+      (config.useCurrentFilters
+        ? currentFilters
+        : currentFilters?.filter(f => esFilters.isFilterPinned(f))) ?? [];
+
+    // if useCurrentDashboardDataRange is enabled, then preserve current time range
+    // if undefined is passed, then destination dashboard will figure out time range itself
+    // for brush event this time range would be overwritten
+    let timeRange = config.useCurrentDateRange ? currentTimeRange : undefined;
+    let filtersFromEvent = await (async () => {
+      // TODO: not sure what would be the best way to handle types here
+      // context.data is `unknown` and comes from `EmbeddableVisTriggerContext`
+      try {
+        return (context.data as any).range
+          ? await createFiltersFromRangeSelectAction(context.data as any)
+          : await createFiltersFromValueClickAction(context.data as any);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn("DashboardToDashboard drilldown: can't extract filters from event", e);
+        return [];
```

Comment:
I think it worth muting the error here? 
If we miss accidentally miss an edge case here caused by the event payload 
* with muted error worst scenario we get is that user won't get all the filters when drilling down. But at least will be successfully navigated to destination dashboard with some filters, query and time range 
* otherwise no navigation happens and more frustration?